### ISSUE TITLE
h3 is now at version 4.0 with breaking API changes

### ISSUE THREAD

**Issue by robfitzgerald** [NREL/hive#139]
noted today that h3 is now at 4.0. the 4.0 API changes the name of methods. straight off the top of the [documentation](https://h3geo.org/docs/api/indexing/), the [3.x geo_to_h3 and h3_to_geo functions](https://h3geo.org/docs/3.x/api/indexing) have been renamed latlng_to_cell and cell_to_latlng.

a quick immediate update is to restrict h3 versions to 3.x in our pyproject.toml, and long-term, we should update to the latest version or provide an abstraction to support both versions.

**Comment by clintonsteiner** [comment]
Would like to move to version 4, import h3
20 files total that need to be updated
rg 'import h3' | wc -l

**Comment by clintonsteiner** [comment]
https://h3geo.org/docs/library/migration-3.x/functions looks like the functions were just renamed, working on updating now

**Comment by clintonsteiner** [comment]
3.7.6 is latest stable version
find . -type f -exec sed -i 's/h3.geo_to_h3/h3.latLngToCell/g' {} \;
Here is the sed command I used to begin updating using the reference above
https://github.com/uber/h3-py

**Comment by nreinicke** [comment]
Considering that version 4.0 isn't yet stable we should probably hold off on including it as a required dependency. Maybe we defer this until 4.0 is the de-facto version?

**Comment by robfitzgerald** [comment]
> 3.7.6 is latest stable version

right. sorry, i only checked the [base library](https://github.com/uber/h3) when i created this ticket. it appears that the python bindings for the core h3 library haven't caught up to the v4.1.0 c lib. and, ya, let's hold off on committing then until 4+ goes stable for python. should we create a temp branch off the hive repo to merge this work into? thanks for your work @clintonsteiner.

**Comment by clintonsteiner** [comment]
Going to have a tough time replacing polyfill, leaving that for your team. 
Only one usage in geofence.py, replacements are cells_to_polygon or polygon_to_cells

**Comment by clintonsteiner** [comment]
https://github.com/NREL/hive/pull/181
Put this in draft status here, until uber h3 goes stable
